ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/machine/WsAgentInitializer.java

```diff
@@ -0,0 +1,35 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.ide.api.machine;
+
+/**
+ * Initialize client's workspace agent component with specific developer machine. Usually, this interface
+ * is not intended to be implemented by the third-party components.
+ *
+ * @author Vlad Zhukovskyi
+ * @see DevMachine
+ * @since 5.0.0
+ */
+public interface WsAgentInitializer {
```

Comment:
As far as I understood this interface is used in IDE app only and not intended to be used by extensions.
So, if it isn't a part of IDE public API why do we need it in IDE API? May be it would be better to move it in IDE app module?